libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1307,19 +1307,26 @@ AP_GPS_UBLOX::broadcast_configuration_failure_reason(void) const {
 /*
   return velocity lag in seconds
  */
-float AP_GPS_UBLOX::get_lag(void) const
+bool AP_GPS_UBLOX::get_lag(float &lag_sec) const
 {
     switch (_hardware_generation) {
+    case UBLOX_UNKNOWN_HARDWARE_GENERATION:
+        lag_sec = 0.22f;
+        // always bail out in this case, it's used to indicate we have yet to recieve a valid
+        // hardware generation
+        return false;
     case UBLOX_5:
     case UBLOX_6:
     default:
-        return 0.22f;
+        lag_sec = 0.22f;
+        break;
     case UBLOX_7:
     case UBLOX_M8:
         // based on flight logs the 7 and 8 series seem to produce about 120ms lag
-        return 0.12f;
+        lag_sec = 0.12f;
         break;
     };
+    return !(_unconfigured_messages & CONFIG_VERSION);
```

Comment:
MON_VER isn't the same. If you look at what we get back from that, we get a 10 character long string, which while we could build a parse table, is less useful then a enumerated return type which is what we are getting out of SVINFO.

Regarding the configure bit, I agree with you, but we also have 2 ways of tracking if hardware generation is present (we can either check the config bitmask, or check the _hardware_generation parameter), and I went with the bitmask. I think a pass is needed to reduce this scope, however this is entirely outside this PR and I've addressed it.